internal/fwserver/server_configureprovider.go

```diff
@@ -22,6 +22,18 @@ func (s *Server) ConfigureProvider(ctx context.Context, req *provider.ConfigureR
 
 	logging.FrameworkTrace(ctx, "Called provider defined Provider Configure")
 
+	if resp.Deferred != nil {
+		if !req.ClientCapabilities.DeferralAllowed {
+			resp.Diagnostics.AddError("Invalid Deferred Provider Response",
+				"Provider configured a deferred response for all resources and data sources but the Terraform request "+
+					"did not indicate support for deferred actions. This is an issue with the provider and should be reported to the provider developers.")
```

Comment:
super nit: We could probably immediately return after this, since the return isn't right after this statement